File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        
        int m = word1.length();
        int n = word2.length();
        
        if(m != n) return false;
        
        vector<int> freq1(26, 0), freq2(26, 0);
        
        for(int i=0; i<m; i++){
            int idx1 = word1[i]-'a';
            int idx2 = word2[i]-'a';
            
            freq1[idx1]++;
            freq2[idx2]++;
        }
        
        for(int i=0; i<26; i++){
            if(freq1[i] != 0 and freq2[i] != 0) continue;
            if(freq1[i] == 0 and freq2[i] == 0) continue;
            return false;
        }
        
        sort(freq1.begin(), freq1.end());
        sort(freq2.begin(), freq2.end());
        
        return freq1 == freq2;
        
    }
};

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        
        if(word1.length() != word2.length()) return false;
        
        int[] freq1 = new int[26];
        int[] freq2 = new int[26];
        
        for(int i=0; i<word1.length(); i++){
            
            int idx1 = word1.charAt(i)-'a';
            int idx2 = word2.charAt(i)-'a';
            
            freq1[idx1]++;
            freq2[idx2]++;
        }
        
        for(int i=0; i<26; i++){
            if(freq1[i] != 0 && freq2[i] != 0) continue;
            if(freq1[i] == 0 && freq2[i] == 0) continue;
            return false;
        }
        
        Arrays.sort(freq1);
        Arrays.sort(freq2);
        
        for(int i=0; i<26; i++){
            if(freq1[i] != freq2[i]) return false;
        }
        
        return true;
    }
}